.eslintrc.json

```diff
@@ -51,19 +56,19 @@
         "rules": [
           {
             "from": "web",
-            "allow": ["web", "interface-adapters", "entities"]
+            "allow": ["web", "interface-adapters", "entities", "ioc"]
           },
           {
             "from": "interface-adapters",
-            "allow": ["use-cases", "entities"]
+            "allow": ["use-cases", "entities", "ioc"]
           },
           {
             "from": "infrastructure",
             "allow": ["service-interfaces", "repository-interfaces", "entities"]
           },
           {
             "from": "use-cases",
-            "allow": ["entities"]
+            "allow": ["entities", "ioc"]
```

Comment:
You should not allow to use the service locator / resolver from everywhere as it will be hard to keep the constraints on the dependency direction rules + it is considered as an anti-pattern normally to use it everywhere. Consider using it at the top level would be better and keep DI elsewhere. So you don't hide the dependencies and avoid mistakes. Wdyt?

Reply:
And pass the services/repos through args? I need them in the use cases because use cases interact with repos & services.

Reply:
You can use the service locator to get your controller (it needs your use case, so the use case will already be build with its own dependencies). Then you can use DI using constructor if it is a class or DI using higher order functions if it is a function. Something like this:

Controller:
```typescript
interface Dependencies {
    authenticationService: IAuthenticationService;
    createTodoUseCase: ICreateTodoUseCase;
}

export const CreateTodoController = ({authenticationService, createTodoUseCase}: Dependencies) => ({
    async createTodo(input: any, sessionId: string | undefined): Promise<ReturnType<typeof presenter>> {
        return await startSpan({name: "createTodo Controller"},
            async () => {
                if (!sessionId) {
                    throw new UnauthenticatedError("Must be logged in to create a todo");
                }
                const {user} = await authenticationService.validateSession(sessionId);

                const {data, error: inputParseError} = inputSchema.safeParse(input);

                if (inputParseError) {
                    throw new InputParseError("Invalid data", {cause: inputParseError});
                }

                const todo = await createTodoUseCase.execute(data, user.id);

                return presenter(todo);
            },
        );
    }
});
```

and Use case:
```typescript
interface Dependencies {
    todosRepository: ITodosRepository;
}

export const CreateTodoUseCase = ({todosRepository}: Dependencies) => ({
    async execute(input: { todo: string }, userId: string): Promise<Todo> {
        return startSpan(
            {name: "createTodo Use Case", op: "function"},
            async () => {
                // HINT: this is where you'd do authorization checks - is this user authorized to create a todo
                // for example: free users are allowed only 5 todos, throw an UnauthorizedError if more than 5

                return await todosRepository.createTodo({
                    todo: input.todo,
                    userId,
                    completed: false,
                });
            },
        );
    }
});
```

Reply:
That's actually pretty smart! Put use cases under IoC and instantiate them with the dependencies included. I'll def look into this. Thanks!

Reply:
Done! Really like how this looks like with `ioctopus`.